packages/react-components/react-jsx-runtime/src/createElementNext.ts

```diff
@@ -0,0 +1,33 @@
+import * as React from 'react';
+import { isSlot, UnknownSlotProps, SLOT_COMPONENT_METADATA_SYMBOL, SlotComponent } from '@fluentui/react-utilities';
+
+export function createElementNext<Props extends {}>(
+  type: React.ElementType<Props>,
+  props?: Props | null,
+  ...children: React.ReactNode[]
+): React.ReactElement<Props> | null {
+  return isSlot(type) ? createElementFromSlotComponent(type, children) : React.createElement(type, props, ...children);
+}
+
+function createElementFromSlotComponent<Props extends UnknownSlotProps>(
+  type: SlotComponent<Props>,
+  overrideChildren: React.ReactNode[],
+): React.ReactElement<Props> | null {
```

Comment:
It looks like createElementFromSlotComponent is ignoring any props set in JSX. Setting slot props in JSX is definitely not recommended, but since I don't think we can prevent it via TypeScript, then we should probably handle it here. Otherwise it will silently fail to work, and could be very confusing to a dev.

I.e. this should work, and result in the `foo` prop being overridden to be `"bar"`:
```jsx
<state.mySlot foo="bar" />
```

Reply:
> Setting slot props in JSX is definitely not recommended, but since I don't think we can prevent it via TypeScript, then we should probably handle it here. Otherwise it will silently fail to work, and could be very confusing to a dev.

Yes, we can prevent it via Typescript!